src/App.tsx

```diff
@@ -1,61 +1,121 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoWithUser } from './types/Todo';
+import { TodoList } from './components/TodoList';
+
+const preperedTodos: TodoWithUser[] = todosFromServer.map(todo => {
+  const user = usersFromServer.find(({ id }) => id === todo.userId);
+
+  return {
+    ...todo,
+    user,
+  };
+});
 
 export const App = () => {
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [hasUserIdError, setHasUserIdError] = useState(false);
+  const [todos, setTodos] = useState(preperedTodos);
+
+  function handleTitleInput(event: React.ChangeEvent<HTMLInputElement>) {
+    setTitle(event.target.value);
+  }
+
+  function handleSelectUserId(event:React.ChangeEvent<HTMLSelectElement>) {
+    setUserId(+event.target.value);
+  }
+
+  function handleOnAdd(event: React.FormEvent<HTMLFormElement>) {
+    event.preventDefault();
+
+    const user = usersFromServer.find(({ id }) => id === userId);
```

Comment:
You search for a user by ID several times, this can be carried out in a separate helper.